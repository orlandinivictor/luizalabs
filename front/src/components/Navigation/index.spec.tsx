import { describe, expect, it, vi } from "vitest";
import { render, fireEvent, screen } from "@testing-library/react";
import renderer from "react-test-renderer";

import { Navigation } from ".";

describe("Navigation Tests", () => {
  it("match snapshot", () => {
    const navigate = vi.fn();

    const domTree = renderer
      .create(<Navigation navigate={navigate} />)
      .toJSON();

    expect(domTree).toMatchSnapshot();
  });

  it("should be call navigate function", () => {
    const navigate = vi.fn();

    render(<Navigation navigate={navigate} pathname="wishlist" />);
    fireEvent.click(screen.getByText("Home"));

    expect(navigate).toBeCalled();
    expect(navigate).toBeCalledWith("/home");
  });
});
