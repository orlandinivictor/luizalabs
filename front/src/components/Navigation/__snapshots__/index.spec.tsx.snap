// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navigation Tests > match snapshot 1`] = `
<nav
  className="sc-braxZu eOGVZE"
>
  <button
    data-content="home"
    onClick={[Function]}
  >
    Home
  </button>
  <span>
    /
  </span>
  <button
    data-content="wishlist"
  >
    Wishlist
  </button>
</nav>
`;
